src/tests/dotnet-counters/CounterMonitorPayloadTests.cs

```diff
@@ -0,0 +1,311 @@
+                        processId: testRunner.Pid,
+                        refreshInterval: 1,
+                        format: exportFormat,
+                        output: path,
+                        name: null,
+                        diagnosticPort: null,
+                        resumeRuntime: false,
+                        maxHistograms: 10,
+                        maxTimeSeries: 1000,
+                        duration: TimeSpan.FromSeconds(10)));
+            }, testRunner, source.Token);
+        }
+
+        private void ValidateCustomMetrics(ISet<string> actualProviders, ISet<string> actualCounterNames, IEnumerable<string> actualCounterTypes, ISet<string> actualHistogramTags, ISet<string> actualCounterTags, List<double> actualHistogramValues, List<double> actualCounterValues, CountersExportFormat format)
+        {
+            // Currently not validating timestamp due to https://github.com/dotnet/diagnostics/issues/3905
+
+            HashSet<string> expectedProviders = new() { Constants.TestMeterName };
+            Assert.Equal(expectedProviders, actualProviders);
+
+            HashSet<string> expectedCounterNames = new() { Constants.TestHistogramName, Constants.TestCounterName };
+            Assert.Equal(expectedCounterNames, actualCounterNames);
+
+            Assert.Equal(ExpectedCounterTypes, actualCounterTypes);
+
+            string tagSeparator = format == CountersExportFormat.csv ? ";" : ",";
+            string tag = Constants.TagKey + "=" + Constants.TagValue + tagSeparator + Constants.PercentileKey + "=";
+            HashSet<string> expectedTags = new() { $"{tag}50", $"{tag}95", $"{tag}99" };
+            Assert.Equal(expectedTags, actualHistogramTags);
+            Assert.Empty(actualCounterTags);
+
+            Assert.Equal(2, actualCounterValues.Distinct().Count());
+            Assert.Equal(1, actualCounterValues.First());
+            Assert.Equal(0, actualCounterValues.Last());
+            double histogramValue = Assert.Single(actualHistogramValues.Distinct());
+            Assert.Equal(10, histogramValue);
+        }
+
+        private ISet<string> GetCSVTags(List<string> countersList, string counterName)
+        {
+            var tags = countersList.Where(counter => counter.Contains(counterName)).Select(counter => {
+                var split = counter.Split(TagStart);
+                return split.Length > 1 ? split[1].Remove(split[1].Length - 1) : string.Empty;
+            }).ToHashSet();
+            tags.Remove(string.Empty);
+
+            return tags;
+        }
+
+        private List<double> GetCSVValues(IEnumerable<string[]> splitLines, string counterName)
+        {
+            return splitLines.Where(line => line[Constants.CounterNameIndex].Contains(counterName)).Select(line => {
+                return double.TryParse(line[Constants.ValueIndex], out double val) ? val : -1;
+            }).ToList();
+        }
+
+        private void CheckFramework(string buildProjectFramework)
```

Comment:
We use enums for this in dotnet-monitor. What does the rest of the diagnostics code base use for framework version checks?

Reply:
Here's an example of them doing something similar https://github.com/dotnet/diagnostics/blob/main/src/tests/Microsoft.Diagnostics.NETCore.Client/GetProcessEnvironmentTests.cs#L46 - I'll update ours to follow that convention.